pkg/snapshot/filesystem.go

```diff
@@ -0,0 +1,782 @@
+	service := fileSystemMinioServiceResource()
+
+	existingService, err := clientset.CoreV1().Services(namespace).Get(ctx, service.Name, metav1.GetOptions{})
+	if err != nil {
+		if !kuberneteserrors.IsNotFound(err) {
+			return errors.Wrap(err, "failed to get existing service")
+		}
+
+		_, err = clientset.CoreV1().Services(namespace).Create(ctx, service, metav1.CreateOptions{})
+		if err != nil {
+			return errors.Wrap(err, "failed to create service")
+		}
+
+		return nil
+	}
+
+	existingService = updateFileSystemMinioService(existingService, service)
+
+	_, err = clientset.CoreV1().Services(namespace).Update(ctx, existingService, metav1.UpdateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to update service")
+	}
+
+	return nil
+}
+
+func fileSystemMinioServiceResource() *corev1.Service {
+	return &corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Service",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: FileSystemMinioServiceName,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Selector: map[string]string{
+				"app": "kotsadm-fs-minio",
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Protocol:   corev1.ProtocolTCP,
+					Port:       FileSystemMinioServicePort,
+					TargetPort: intstr.FromInt(9000),
+				},
+			},
+		},
+	}
+}
+
+func updateFileSystemMinioService(existingService, desiredService *corev1.Service) *corev1.Service {
+	existingService.Spec.Ports = desiredService.Spec.Ports
+
+	return existingService
+}
+
+func shouldResetFileSystemMount(ctx context.Context, clientset kubernetes.Interface, deployOptions FileSystemDeployOptions, registryOptions kotsadmtypes.KotsadmOptions) (shouldReset bool, hasMinioConfig bool, finalErr error) {
+	checkPod, err := createFileSystemMinioCheckPod(ctx, clientset, deployOptions, registryOptions)
+	if err != nil {
+		finalErr = errors.Wrap(err, "failed to create file system minio check pod")
+		return
+	}
+
+	if err := k8sutil.WaitForPodCompleted(ctx, clientset, deployOptions.Namespace, checkPod.Name, time.Minute*2); err != nil {
+		finalErr = errors.Wrap(err, "failed to wait for file system minio check pod to complete")
+		return
+	}
+	defer clientset.CoreV1().Pods(deployOptions.Namespace).Delete(ctx, checkPod.Name, metav1.DeleteOptions{})
+
+	logs, err := k8sutil.GetPodLogs(ctx, clientset, checkPod)
+	if err != nil {
+		finalErr = errors.Wrap(err, "failed to get file system minio check pod logs")
+		return
+	}
+	if len(logs) == 0 {
+		finalErr = errors.New("no logs found")
+		return
+	}
+
+	type FileSystemMinioCheckPodOutput struct {
+		HasMinioConfig bool   `json:"hasMinioConfig"`
+		MinioKeysSHA   string `json:"minioKeysSHA"`
+	}
+
+	checkPodOutput := FileSystemMinioCheckPodOutput{}
+
+	scanner := bufio.NewScanner(bytes.NewReader(logs))
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if err := json.Unmarshal([]byte(line), &checkPodOutput); err != nil {
```

Comment:
Reusing the same variable to unmarshal in a loop is not totally safe.  I guess I'd be curious to see what we are parsing here.